### ISSUE TITLE
error: tls: server selected unsupported group

### ISSUE THREAD

**Issue by 89z** [Danny-Dasilva/CycleTLS#43]
Using the JA3 from here [1], and this program:

~~~go
package main

import (
   "fmt"
   "github.com/Danny-Dasilva/CycleTLS/cycletls"
)

const test = "771,4865-4867-4866-49195-49199-52393-52392-49196-49200-49162-49161-49171-49172-51-57-47-53-10,0-23-65281-10-11-35-16-5-51-43-13-45-28-21,29-23-24-25-256-257,0"

func main() {
   opt := cycletls.Options{Ja3: test}
   res, err := cycletls.Init().Do("https://example.com", opt, "HEAD")
   if err != nil {
      panic(err)
   }
   fmt.Println(res.Response.Body)
}
~~~

I get this result:

~~~
->
Head "https://example.com": uTlsConn.Handshake() error: tls: server selected
unsupported group
~~~

1. https://github.com/Danny-Dasilva/CycleTLS#single-request-example-for-tsjs

**Comment by Danny-Dasilva** [comment]
This issue is actually solved by [your fix on mytls](https://github.com/zedd3v/mytls/issues/7). Passing the correct TLS version for the extensions is the way to go. Thanks for the contribution, this fix should be in the next release.

**Comment by Danny-Dasilva** [comment]
Issue caused by the incorrect usage of the `supported_versions` extension (43). If you find any issues in the Repo or have any feature suggestions feel free to reach out.

**Comment by 89z** [comment]
can you clarify? Seems to be an issue with this package, not what you said.

**Comment by Danny-Dasilva** [comment]
@89z the example.com url was failing because the `supported_versions` extension (extension number 43 was not set with the proper value). This was value not being properly set thus causing the `uTlsConn.Handshake() error: tls: server selected
unsupported group` error

I updated the CycleTLS package to properly handoff `SSL` version to the `supported_versions` extension. This allows the `example.com` to do the TLS handshake correctly and return a 200 response. 

https://github.com/Danny-Dasilva/CycleTLS/blob/c5dc8d95287911b2059e7bb27976ac6c64221699/cycletls/roundtripper.go#L239-L243


You are correct the issue is with this package I just applied the solution you suggested for mytls to this repository.
